src/backup_restore.cpp

```diff
@@ -1,16 +1,155 @@
 #include "backup_restore.hpp"
 
+#include "constants.hpp"
+#include "exceptions.hpp"
+#include "logger.hpp"
+#include "parser.hpp"
+#include "utils.hpp"
+
 namespace vpd
 {
-BackupRestore::BackupRestore(const nlohmann::json& i_parsedSystemJson)
+BackupRestore::BackupRestore(const nlohmann::json& i_sysCfgJsonObj) :
+    m_sysCfgJsonObj(i_sysCfgJsonObj)
 {
-    (void)i_parsedSystemJson;
+    try
+    {
+        if (const std::string& l_backupRestoreCfgJsonFilePath =
+                i_sysCfgJsonObj.value("backupRestoreConfigPath", "");
+            !l_backupRestoreCfgJsonFilePath.empty())
+        {
+            m_backupRestoreCfgJsonObj =
+                utils::getParsedJson(l_backupRestoreCfgJsonFilePath);
+        }
+    }
+    catch (const std::exception& ex)
+    {
+        logging::logMessage(
+            "Failed to parse backup and restore config JSON file, exception: " +
+            std::string(ex.what()));
+    }
 }
 
 std::tuple<types::VPDMapVariant, types::VPDMapVariant>
     BackupRestore::backupAndRestore()
 {
-    return std::make_tuple(std::monostate{}, std::monostate{});
+    auto l_emptyVariantPair = std::make_tuple(std::monostate{},
+                                              std::monostate{});
+    if (m_backupAndRestoreDone)
+    {
+        logging::logMessage("Backup and restore triggered already.");
+        return l_emptyVariantPair;
+    }
+
+    try
+    {
+        if (m_backupRestoreCfgJsonObj.empty() ||
+            !m_backupRestoreCfgJsonObj.contains("source") ||
+            !m_backupRestoreCfgJsonObj.contains("destination") ||
+            !m_backupRestoreCfgJsonObj.contains("type") ||
+            !m_backupRestoreCfgJsonObj.contains("backupMap"))
+        {
+            logging::logMessage(
+                "Backup restore config JSON is missing necessary tag(s), can't initiate backup and restore.");
+            return l_emptyVariantPair;
+        }
+
+        std::string l_srcVpdPath;
+        types::VPDMapVariant l_srcVpdMap;
+        if (const std::string& l_srcHwFilePath =
+                m_backupRestoreCfgJsonObj["source"].value("hardwarePath", "");
+            !l_srcHwFilePath.empty() &&
+            std::filesystem::exists(l_srcHwFilePath))
+        {
+            std::shared_ptr<Parser> l_vpdParser =
+                std::make_shared<Parser>(l_srcHwFilePath, m_sysCfgJsonObj);
+            l_srcVpdMap = l_vpdParser->parse();
+            l_srcVpdPath = l_srcHwFilePath;
+        }
+        else if (const std::string& l_srcInventoryObjPath =
```

Comment:
> l_srcInventoryObjPath

We can get rid of this variable and a string copy after that by using l_srcVpdPath directly?

Reply:
Done